controller/controller.go

```diff
@@ -314,32 +322,6 @@ func getProvider(providers driver.TerraformProviderBlocks, id string) driver.Ter
 	})
 }
 
-// getTemplateBufferPeriod applies the task buffer period config to its template
-func getTemplateBufferPeriod(conf *config.Config,
```

Comment:
The refactor to resolve the buffer period earlier is a good improvement and helps with clarity and expectation by the time `config.Config` is passed to the controller.